code/Model/SiteTree.php

```diff
@@ -1604,6 +1604,21 @@ public function generateURLSegment($title)
         // Hook for extensions
         $this->extend('updateURLSegment', $t, $title);
 
+        $excludes = Director::config()->get('rules');
+        $excludes = array_keys($excludes);
+        $disallowed = array_map(function ($key) {
```

Comment:
You can also use `strtok($key, '/');` as shorthand too. :) (not a change request, just an aside).